File: ruby/basic/num_palindrome_check.rb
loop do
  puts "Enter a Number or (q)uit:"
  input = gets.chomp
  break if input == 'q'

  original = input.to_i
  num = original
  reverse = 0
  while num > 0
    tmp = num % 10
    num = (num - tmp)/10
    reverse = reverse * 10 + tmp
  end
  if reverse == original
    puts "Num is palindrome"
  else
    puts "Num is not palindrome"
  end
end

File: c/basic/num_palindrome_check.c
/*
 * Compile With:
 *   gcc -Wall -o num_palindrome_check num_palindrome_check.c
 * Run: ./num_palindrome_check
 */

#include <stdio.h>

char QUIT = 'q';

int main(){
  int num;
  int newNumber = 0;
  int originalNumber;
  char string[100];
  char quit;
  
  while(1) {
    printf("Enter a Number:");
    fgets(string, 100, stdin);
    
    sscanf(string, "%c", &quit);
    if (quit == QUIT) {
      printf("Goodbye\n");
      return 0;
    } else {

      sscanf(string, "%d", &originalNumber);
      printf("You entered %d\n", originalNumber);
      num = originalNumber;

      
      while(num > 0) {
        int tmp = num % 10;
        num = (num - tmp)/10;
        newNumber = newNumber * 10 + tmp;
      }

      if (newNumber == originalNumber)
        printf("Number is a Palindrome\n");
      else
        printf("Number is not a Palindrome\n");

      newNumber = 0;
    }
  }
}

